stat_var_renaming/stat_var_renaming.py

```diff
@@ -0,0 +1,631 @@
+      dpv_match = dpv_match.query(f"orig_v{c}_x == orig_v{c}_y")
+      stat_vars.loc[dpv_match.index, f"p{c}"] = np.nan
+      stat_vars.loc[dpv_match.index, f"v{c}"] = np.nan
+      stat_vars.loc[dpv_match.index, "numConstraints"] = (
+        stat_vars.loc[dpv_match.index, "numConstraints"].apply(lambda x: x - 1))
+
+  # Left shift all imputed columns to remove holes.
+  stat_vars = stat_vars.apply(left_fill_columns, axis=1)
+
+  # Rename constraints from merge.
+  for c in range(1, MAX_CONSTRAINTS + 1):
+    stat_vars = stat_vars.rename({f"orig_v{c}_x":f"orig_v{c}"}, axis=1)
+  return stat_vars
+  
+
+def left_fill_columns(row):
+  """ Removes holes in constraints after imputing depedent constraints. 
+    Args:
+      Row of dataframe with or without holes present between constraints.
+    
+    Returns:
+      Row of dataframe without holes present between constraints.
+  """
+  row_constraints = min(row['numConstraints'], MAX_CONSTRAINTS_WITH_DPV)
+  # Keep track of the search location to look to pull columns from. 
+  search_location = 2
+  for base_col in range(1, row_constraints + 1):
+    # If current population is null then search for the next non-null column.
+    if pd.isna(row[f"p{base_col}"]):
+      search_location = max(search_location, base_col + 1)
+      while search_location <= MAX_CONSTRAINTS:
+        # Swap first non-null with the current null location.
+        if not pd.isna(row[f"p{search_location}"]):
+          row[f"p{base_col}"] = row[f"p{search_location}"]
+          row[f"v{base_col}"] = row[f"v{search_location}"]
+          row[f"p{search_location}"] = np.nan
+          row[f"v{search_location}"] = np.nan
+          search_location += 1
+          break
+        search_location += 1
+  return row
+
+
+def row_to_human_readable(row):
+  """ Generates a human readable name for a dataframe row.
+  Args:
+    Row -> A preprocessed dataframe row with dependent variables removed and
+      all constraints, values, and populations remove.
+
+  Returns:
+    String mapping between the provided row to the corresponding human readable
+    dcid format: <?statType>_<mProp>_<popType>_<v1>_<v2>_..._<mQual>_<mDenom>
+  """
+  # Add measured property and population type. e.g. Count_InsuranceClaim.
+  human_string = (f"{capitalizeFirst(row['measuredProp'])}" +
+                  f"_{capitalizeFirst(row['populationType'])}")
+
+  # StatType (e.g. median) is prepending if the stat type is not measuredValue.
+  stat_type = row['statType']
+  if stat_type != "measuredValue" and stat_type != "Unknown":
+    human_string = (capitalizeFirst(stat_type.replace("Value", ""))
+                    + "_" + human_string)
+
+  # Postpend renamed constraint fields.
+  row_constraints = min(row['numConstraints'], MAX_CONSTRAINTS_WITH_DPV)
+  for num in range(1, row_constraints + 1): 
+    if not pd.isna(row[f"v{num}"]):
+      human_string += "_" + row[f"v{num}"]
+
+  # Postpend nmeasurement qualifier if it exists.
```

Comment:
Typically, the word is "append", which is not anti-symmetric with "prepend".

Reply:
Changed.